src/components/Modal/CalendarAlertModal.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useMemo, useState } from 'react'
+
+import { UnitTypeLongPlural } from 'dayjs'
+import { Button } from 'decentraland-ui/dist/components/Button/Button'
+import { Close } from 'decentraland-ui/dist/components/Close/Close'
+import { Header } from 'decentraland-ui/dist/components/Header/Header'
+import { Modal, ModalProps } from 'decentraland-ui/dist/components/Modal/Modal'
+
+import { ProposalAttributes } from '../../entities/Proposal/types'
+import { proposalUrl } from '../../entities/Proposal/utils'
+import useFormatMessage from '../../hooks/useFormatMessage'
+import Time from '../../utils/date/Time'
+import { getGoogleCalendarUrl } from '../../utils/projects'
+import NumberSelector from '../ProjectRequest/NumberSelector'
+
+import './CalendarAlertModal.css'
+
+type CalendarAlertModalProps = Omit<ModalProps, 'children'> & {
+  proposal: Pick<ProposalAttributes, 'id' | 'title' | 'finish_at'>
+}
+
+const UNITS: UnitTypeLongPlural[] = ['seconds', 'minutes', 'hours', 'days']
+
+function getAlertDate(finishAt: ProposalAttributes['finish_at'], value: number, unit: UnitTypeLongPlural): Date | null {
+  const proposalFinishDate = Time.from(finishAt)
+  const currentDate = Time()
+  const alertDate = proposalFinishDate.subtract(value, unit)
+  if (alertDate.isBefore(currentDate)) {
+    return null
+  }
+  return alertDate.toDate()
+}
+
+function CalendarAlertModal({ proposal, onClose, ...props }: CalendarAlertModalProps) {
+  const t = useFormatMessage()
+  const { finish_at, title, id } = proposal
+  const [value, setValue] = useState(0)
+  const [unit, setUnit] = useState<UnitTypeLongPlural | undefined>()
+  const [isDisabled, setIsDisabled] = useState(false)
+
+  const handleOnClose = () => {
+    onClose()
+    setValue(0)
+    setUnit(undefined)
+  }
+
+  const alertDate = useMemo(() => (unit ? getAlertDate(finish_at, value, unit) : null), [finish_at, unit, value])
+
+  useEffect(() => {
+    setIsDisabled(!alertDate)
+  }, [alertDate])
+
+  const alertUrl = getGoogleCalendarUrl({
+    title: t(`modal.calendar_alert.calendar_title`, { title }),
+    details: t(`modal.calendar_alert.calendar_details`, {
+      finish_date: Time.from(finish_at).format('MMM DD, YYYY HH:mm'),
+      url: proposalUrl(id),
+    }),
+    startAt: alertDate!,
+  })
+
+  return (
+    <Modal {...props} onClose={handleOnClose} size="tiny" closeIcon={<Close />}>
+      <Modal.Content>
+        <div className="ProposalModal__Title">
+          <Header>{t('modal.calendar_alert.title')}</Header>
+        </div>
+        <div className="CalendarAlertModal__SelectorContainer">
+          <NumberSelector
+            value={value}
+            min={0}
+            max={300}
```

Comment:
what is 300? 

<img width="455" alt="image" src="https://github.com/decentraland/governance/assets/2858950/ae668946-6436-47ef-97e2-d5cf23ee5f52">